geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GridSnap.scala

```diff
@@ -97,13 +98,18 @@ class GridSnap(env: Envelope, xSize: Int, ySize: Int) {
         }
         def hasNext = stepX * xT <= fX && stepY * yT <= fY
       }
-      iter.toList.dropRight(1).toSet
+      iter.toList.dropRight(1).distinct
     }
   }
 
   /** Generate a Set of Coordinates between two given Snap Coordinates which includes both start and end points*/
   def generateLineCoordSet(coordOne: Coordinate, coordTwo: Coordinate): Set[Coordinate] = {
-    genBresenhamCoordSet(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)) + coordOne + coordTwo
+    genBresenhamCoordList(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)).toSet + coordOne + coordTwo
+  }
+
+  /** Generate an ordered List of Coordinates between two given Snap Coordinates which includes both start and end points*/
+  def generateLineCoordList(coordOne: Coordinate, coordTwo: Coordinate): List[Coordinate] = {
+    coordOne +: genBresenhamCoordList(i(coordOne.x), j(coordOne.y), i(coordTwo.x), j(coordTwo.y)) :+ coordTwo
```

Comment:
is there any possibility of coordOne or two being duplicated? In the 'generateLineCoordSet' method the set type will prevent that.